db/single/reference/webp/generating.py

```diff
@@ -0,0 +1,59 @@
+#!/bin/python3
+
+# Copyright (c) 2022, NVIDIA CORPORATION & AFFILIATES. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from PIL import Image
+import numpy as np
+import cv2
+
+# https://en.wikipedia.org/wiki/YCbCr#ITU-R_BT.601_conversion
+def rgb_to_ycbcr(rgb):
+    ycbcr_weights = np.asarray([[ 0.257,  0.504,  0.098],
+                                [-0.148, -0.291,  0.439],
+                                [ 0.439, -0.368, -0.071]]).T
+    ycbcr = np.matmul(rgb, ycbcr_weights) + np.asarray([0.0625, 0.5, 0.5])
+    return ycbcr
+
+def to_uint8(float_image):
+    return np.round((float_image * 255) + 1).astype('uint8')
+
+def get_ycbcr(image):
+    rgb = np.asarray(image).astype('float32') / 255
+    ycbcr = rgb_to_ycbcr(rgb)
+    return to_uint8(ycbcr)
```

Comment:
I know this is just to generate reference, but I saw this repeated several times already, maybe it's time to place it somewhere once?

Reply:
I see your point, but I don't think it's worth putting the effort now.